drivers/eeprom/Kconfig.mb85rcxx

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config EEPROM_MB85RCXX
+	bool "FUJITSU mb85rcxx i2c FRAM"
+	default y
+	depends on DT_HAS_FUJITSU_MB85RCXX_ENABLED
+	select I2C
+	help
+	  Enable FUJITSU mb85rcxx i2c FRAM
+
+config EEPROM_MB85RCXX_INIT_PRIORITY
+	int "MB85RCXX FRAM init priority"
+	default 80
+	depends on EEPROM_MB85RCXX
+	help
+	  MB85RCXX FRAM driver device initialization priority
```

Comment:
Please do not introduce a driver-specific initialization priority Kconfig. Use `CONFIG_EEPROM_INIT_PRIORIT`Y` but adjust the test, if needed, to ensure the test GPIO controller is initialized before the FRAM driver.

Reply:
Changing the default `CONFIG_EEPROM_INIT_PRIORITY` in `tests/drivers/build_all/eeprom/prj.conf` will be a good solution here?

Reply:
I am unsure why this is even needed. The existing `tests/drivers/build_all/eeprom/` also depends on GPIOs, and it does not fail in priority checking.

The relevant priorities, when running on `native_posix`, are as follows:
```
$ grep -E '(EEPROM|GPIO)_INIT_PRIO' zephyr/.config 
CONFIG_EEPROM_INIT_PRIORITY=50
CONFIG_GPIO_INIT_PRIORITY=40
```

Reply:
@henrikbrixandersen The problem appeared in the CI build: https://github.com/zephyrproject-rtos/zephyr/actions/runs/5976393051/job/16215196325?pr=61653#step:13:1266
But I don't know why, I am not able to reproduce it locally. 
For `em_starterkit_em7d` `tests/drivers/build_all/eeprom/drivers.eeprom.build` passes and the priorities are as follows:
```
grep -E '(EEPROM|GPIO)_INIT_PRIO' build/zephyr/.config
CONFIG_EEPROM_INIT_PRIORITY=50
CONFIG_GPIO_INIT_PRIORITY=70
```
Building with `west` or testing with `twister` works for me without error.

Reply:
Ahh, that is another issue. It is solved here: https://github.com/zephyrproject-rtos/zephyr/pull/61951

Reply:
Great, we will wait until it is merged to main

Reply:
Heya, merged, should be fine to drop this now.

Reply:
Thanks!
@henrikbrixandersen We've dropped the priority config changes.